### PR TITLE
EIRH-55 | @rebeccahongsf | structure filter data

### PR DESCRIPTION
**Pull request by rebeccahongsf**
# READY FOR REVIEW

# Summary
- structure filter data
- configure redux state management for filter and algolia data
- build out filter chips
- build out filter functionality

# Review By (Date)
- retroreview

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Rebuild Cache and import config `drush cr ; drush ci`
3. Navigate to...
4. Verify...

# Associated Issues and/or People
- EIRH-45
- EIRH-55
- EIRH-50
- EIRH-58
- EIRH-54
- EIRH-56
- EIRH-48

### CODE REVIEW

File: src/eventshub/src/utilities/algoliaFiltersData.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      feature_group: group,
+      event_feature_group: eventFeatureGroup,
+    };
+  });
+}
+
+export const algoliaFilterData = async (): Promise<GroupedDataItem[]> => {
+  const algoliaData = await fetchAlgoliaData();
+  const filterData = restructureData(algoliaData as unknown as DataItem[]);
+  return filterData;
```
Comment (rebeccahongsf) [inline src/eventshub/src/utilities/algoliaFiltersData.ts:78]:
```[
  {
    "feature_group": "Food, Drinks, or Items",
    "event_feature_group": [
      {
        "event_feature": "Food Truck at Event",
        "venues": "Uncategorized, Field(s), Outdoor Space",
        "vendors": ["Food/Water Trucks"],
        "policies": ["Food", "Permits", "Catering and Alcohol"]
      },
      {
        "event_feature": "Alcohol",
        "venues": "Uncategorized",
        "vendors": ["Alcohol Service"],
        "policies": ["Alcohol", "Catering and Alcohol", "Licensing", "Permits"]
      },
      {
        "event_feature": "Selling Items at Event",
        "venues": "Uncategorized",
        "vendors": ["Design/Print", "Merchandise", "Graphic Designer"],
        "policies": ["Retail", "Budgeting and Purchasing"]
      },
      {
        "event_feature": "Tabling at Event",
        "venues": "Uncategorized",
        "vendors": ["Merchandise", "Design/Print", "Graphic Designer"],
        "policies": ["Retail"]
      },
      {
        "event_feature": "Swag at Event",
        "venues": "Uncategorized, Conference Center, Performance Space",
        "vendors": ["Merchandise", "Design/Print", "Graphic Designer"],
        "policies": [
          "Retail",
          "Promotion",
          "Budgeting and Purchasing",
          "Marketing"
        ]
      },
      {
        "event_feature": "Food",
        "venues": "Uncategorized",
        "vendors": ["Catering/Food Service", "Food/Water Trucks"],
        "policies": ["Food", "Catering and Alcohol"]
      }
    ]
  },
  {
    "feature_group": "Other",
    "event_feature_group": [
      {
        "event_feature": "Tenting",
        "venues": "Outdoor Space, Field(s), Uncategorized",
        "vendors": ["Stanford Support", "Equipment Rentals"],
        "policies": ["Outdoor", "Use of Campus Space"]
      },
      {
        "event_feature": "Speaker/Presentation",
        "venues": "Uncategorized",
        "vendors": ["Photography/Videography"],
        "policies": ["Special Visitor Support", "Special Events"]
      },
      {
        "event_feature": "Involves Minors",
        "policies": ["Risk Management and Liability", "Security and Safety"]
      },
      {
        "event_feature": "Amplified Sound",
        "venues": "Performance Hall, Performance Space, Amphitheater, Uncategorized",
        "vendors": [
          "Stanford Support",
          "Musicians/Entertainers",
          "Entertainment Rentals",
          "Equipment Rentals"
        ],
        "policies": [
          "Use of Campus Space",
          "Permits",
          "Security and Safety",
          "Performance"
        ]
      },
      {
        "event_feature": "Team Activity",
        "venues": "Outdoor Space",
        "vendors": [
          "Alcohol Service",
          "Catering/Food Service",
          "Activity",
          "Party/Game/Entertainment Rentals",
          "Entertainment Rentals",
          "Transportation"
        ]
      },
      {
        "event_feature": "Fundraising",
        "venues": "Uncategorized",
        "vendors": [
          "Catering/Food Service",
          "Graphic Designer",
          "Equipment Rentals"
        ],
        "policies": ["Fundraising/Charity", "Budgeting and Purchasing"]
      },
      {
        "event_feature": "Ticketing, Registration, or Charging Admission",
        "venues": "Auditorium/Auditoria, Performance Hall, Performance Space",
        "vendors": ["Stanford Support"],
        "policies": [
          "Ticketing",
          "Date Selection",
          "Marketing",
          "Use of Campus Space",
          "Budgeting and Purchasing"
        ]
      }
    ]
  },
  {
    "feature_group": "Size",
    "event_feature_group": [
      {
        "event_feature": "Large Venue (500-1,000)",
        "venues": "Ballroom, Uncategorized"
      },
      {
        "event_feature": "X-Large Venue (1,000+)",
        "venues": "Amphitheater, Auditorium/Auditoria, Stadiums/Stadium, Field(s), Uncategorized"
      },
      {
        "event_feature": "Medium Venue (100-500)",
        "venues": "Uncategorized"
      },
      {
        "event_feature": "Small Venue (Less than 100)",
        "venues": "Classroom(s), Meeting Room(s), Studio(s), Patio(s), Lounge"
      }
    ]
  },
  {
    "feature_group": "Support",
    "event_feature_group": [
      {
        "event_feature": "Needs Additional Equipment ",
        "venues": "Uncategorized",
        "vendors": [
          "Equipment Rentals",
          "Stanford Support",
          "Entertainment Rentals",
          "Party/Game/Entertainment Rentals"
        ],
        "policies": ["Permits", "Budgeting and Purchasing"]
      },
      {
        "event_feature": "Needs Lodging Support ",
        "venues": "Uncategorized",
        "vendors": ["Stanford Support"],
        "policies": ["Hospitality"]
      },
      {
        "event_feature": "Involves Photography or Videography",
        "venues": "Uncategorized",
        "vendors": ["Photography/Videography"],
        "policies": ["Media"]
      },
      {
        "event_feature": "Needs Volunteer/Staff Support",
        "venues": "Uncategorized",
        "vendors": ["Stanford Support"],
        "policies": ["Permits", "Use of Campus Space"]
      },
      {
        "event_feature": "A/V Support Needed",
        "venues": "Uncategorized",
        "vendors": ["Stanford Support", "Equipment Rentals"],
        "policies": ["Media", "Use of Campus Space"]
      },
      {
        "event_feature": "Needs Security Support",
        "venues": "Uncategorized",
        "vendors": ["Stanford Support"],
        "policies": ["Security and Safety", "Risk Management and Liability"]
      },
      {
        "event_feature": "Needs Parking or Transportation Support",
        "venues": "Uncategorized",
        "vendors": ["Transportation", "Stanford Support", "Food/Water Trucks"],
        "policies": ["Transportation", "Parking"]
      }
    ]
  },
  {
    "feature_group": "Indoor/Outdoor/Mixed",
    "event_feature_group": [
      {
        "event_feature": "Outdoor Venue",
        "venues": "Outdoor Space, Terrace, Amphitheater, Sculpture Garden, Patio(s), Uncategorized, Stadiums/Stadium, Plaza, Garden(s), Cafe, Performance Space, Field(s), Exercise Space",
        "vendors": [
          "Party/Game/Entertainment Rentals",
          "Equipment Rentals",
          "Entertainment Rentals"
        ],
        "policies": ["Outdoor", "Use of Campus Space"]
      },
      {
        "event_feature": "Indoor Venue",
        "venues": "Uncategorized, Auditorium/Auditoria, Seminar Room, Ballroom, Classroom(s), Conference Room, Dining Room, Living Room, Meeting Room(s), Hall, Lecture Hall, Performance Hall, Reception Hall, Studio(s), Church, Lounge, Foyer, Gallery, Indoor Space, Lobby",
        "vendors": ["Stanford Support"],
        "policies": ["Use of Campus Space"]
      }
    ]
  }
]
```

@sherakama thoughts on this filter data restructure? 🤔

Comment (sherakama) [reply src/eventshub/src/utilities/algoliaFiltersData.ts:78]: This looks fine to me but I've been thinking about the sorting... There is currently no way to tell which item is where in the Airtable document. I've added sorting (Asc) by default to the index so at least it is alphabetical and won't change/re-order by relevancy on each fetch.

We may have to add a weight column...

Comment (sherakama) [reply src/eventshub/src/utilities/algoliaFiltersData.ts:78]: I'll add it to the set of questions for the airtble content authors

Comment (sherakama) [reply src/eventshub/src/utilities/algoliaFiltersData.ts:78]: It's looking good. I've added a weight field in the data and put the default sorting by it.

Comment (rebeccahongsf) [reply src/eventshub/src/utilities/algoliaFiltersData.ts:78]: Thanks @sherakama! I've updated the restructure data function to sort the filter items by the weight defined in the Airtable 😃 

I'll be moving onto the filter state management and algolia queries today unless there are any additional adjustments to the restructured data we'd like to make 🤔

File: src/eventshub/src/utilities/algoliaFiltersData.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+// "venues": ["Auditorium/Auditoria", "Performance Hall", "Performance Space"],
+const splitIfNecessary = (value: string | string[]): string[] => {
+  if (Array.isArray(value)) {
+    return value.flatMap(item => item.split(', '));
+  } else if (typeof value === 'string') {
+    return value.split(', ');
+  } else {
+    return [];
+  }
+};
```
Comment (sherakama) [inline src/eventshub/src/utilities/algoliaFiltersData.ts]: That is very nice of you to do this but we shouldn't. This is a data format issue and it needs to change upstream.

**Pull request merged**